src/pybind/mgr/dashboard_v2/controllers/rbd_mirroring.py

```diff
@@ -0,0 +1,314 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import json
+import re
+
+from functools import partial
+
+import cherrypy
+import rbd
+
+from ..services.ceph_service import CephService
+from ..tools import ApiController, AuthRequired, BaseController, ViewCache
+from .. import logger
+
+
+@ApiController('rbdmirror')
+@AuthRequired()
+class RbdMirror(BaseController):
+
+    def __init__(self):
+        self.pool_data = {}
+
+    @cherrypy.expose
+    @cherrypy.tools.json_out()
+    def default(self):
+        status, content_data = self._get_content_data()
+        return {'status': status, 'content_data': content_data}
+
+    @ViewCache()
+    def _get_daemons_and_pools(self):  # pylint: disable=R0915
+        def get_daemons():
+            daemons = []
+            for hostname, server in CephService.get_service_map('rbd-mirror').items():
+                for service in server['services']:
+                    id = service['id']  # pylint: disable=W0622
+                    metadata = service['metadata']
+                    status = service['status']
+
+                    try:
+                        status = json.loads(status['json'])
+                    except ValueError:
+                        status = {}
+
+                    instance_id = metadata['instance_id']
+                    if id == instance_id:
+                        # new version that supports per-cluster leader elections
+                        id = metadata['id']
+
+                    # extract per-daemon service data and health
+                    daemon = {
+                        'id': id,
+                        'instance_id': instance_id,
+                        'version': metadata['ceph_version'],
+                        'server_hostname': hostname,
+                        'service': service,
+                        'server': server,
+                        'metadata': metadata,
+                        'status': status
+                    }
+                    daemon = dict(daemon, **get_daemon_health(daemon))
+                    daemons.append(daemon)
+
+            return sorted(daemons, key=lambda k: k['instance_id'])
+
+        def get_daemon_health(daemon):
+            health = {
+                'health_color': 'info',
+                'health': 'Unknown'
+            }
+            for _, pool_data in daemon['status'].items():
```

Comment:
It may make sense to break out of the loop if an error or warning has been found (and set).

Reply:
I'm not really sure if this will change the logic here?

Reply:
Oh, no... I will change it... Forget what I wrote before...